lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py

```diff
@@ -0,0 +1,429 @@
+            ),
+            ip_address=dict(
+                type='str',
+                default='none',
+                choices=['public', 'none']
+            ),
+            ports=dict(
+                type='list',
+                default=[]
+            ),
+            registry_login_server=dict(
+                type='str',
+                default=None
+            ),
+            registry_username=dict(
+                type='str',
+                default=None
+            ),
+            registry_password=dict(
+                type='str',
+                default=None,
+                no_log=True
+            ),
+            containers=dict(
+                type='list',
+                required=True
+            ),
+            force_update=dict(
+                type='bool',
+                default=False
+            ),
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.state = None
+        self.ip_address = None
+
+        self.containers = None
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+
+        super(AzureRMContainerInstance, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                       supports_check_mode=True,
+                                                       supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()):
+            setattr(self, key, kwargs[key])
+
+        resource_group = None
+        response = None
+        results = dict()
+        to_be_updated = False
+
+        self.mgmt_client = self.get_mgmt_svc_client(ContainerInstanceManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group)
+        except CloudError:
+            self.fail('resource group {0} not found'.format(self.resource_group))
+        if not self.location:
+            self.location = resource_group.location
+
+        response = self.get_containerinstance()
+
+        if not response:
+            self.log("Container Group doesn't exist")
+
+            if self.state == 'absent':
+                self.log("Nothing to delete")
+            else:
+                to_be_updated = True
+        else:
+            self.log("Container instance already exists")
+
+            if self.state == 'absent':
+                if not self.check_mode:
+                    self.delete_containerinstance()
+                self.results['changed'] = True
+                self.log("Container instance deleted")
+            elif self.state == 'present':
+                self.log("Need to check if container group has to be deleted or may be updated")
+                to_be_updated = self.force_update
```

Comment:
While this is just nitpicking, why set this to another variable where you can just check `self.force_update`, they are the same variable.